### PR TITLE
fix: Judge stats should update on judge add (#176)

### PR DESCRIPTION
**Pull request by zaheer-Khan3260**
fix: Judge stats should update on judge add (#176 )

### CODE REVIEW

**Comment by zaheer-Khan3260** [comment]
@MichaelZhao21 check this and merge this request it resolve the issue #176

File: client/package-lock.json
Comment (MichaelZhao21) [inline client/package-lock.json:1]: You should not create a package-lock.json; please remove this file.

File: client/yarn.lock
Comment (MichaelZhao21) [inline client/yarn.lock:1]: This file also shouldn't be changed; please revert these changes.

File: client/src/pages/admin/AddJudges.tsx [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
+        setIsJudgeAdded(true);
+    }
+
     return (
         <>
             <JuryHeader withBack withLogout isAdmin />
             <div className="flex flex-col items-start justify-center w-full px-8 py-4 md:px-16 md:py-8">
                 <h1 className="text-4xl font-bold">Add Judges</h1>
-                <AddJudgeStatsPanel />
+                <AddJudgeStatsPanel key={isJudgeAdded.toString()} />
```
Comment (MichaelZhao21) [inline client/src/pages/admin/AddJudges.tsx:19]: What does this achieve? Adding a key does nothing to the component.

File: client/src/pages/admin/AddJudges.tsx [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
                 <div className="mt-8 flex flex-col w-full space-y-8">
-                    <NewJudgeForm />
+                    <NewJudgeForm onAddJudge={handleAddJudge} />
```
Comment (MichaelZhao21) [inline client/src/pages/admin/AddJudges.tsx:21]: Should not be passing a callback to the component when we have a global store.

**Review by MichaelZhao21 — CHANGES_REQUESTED** [review]
This does not solve the problem at hand. You should not be passing callbacks to children. The stats also do not update with these changes. The only change that was needed was a call to the `fetchJudges` function in the Zustand store. There is no need to add more code than one line. You should also not be changing or adding any of the lockfiles.

Furthermore, you did not use the provided pull request template. This will not be accepted in the future.

File: client/src/pages/admin/AddJudges.tsx (hunk 1 above)
Comment (zaheer-Khan3260) [reply client/src/pages/admin/AddJudges.tsx:19]: it can redender the component when the judge is added

File: client/package-lock.json
Comment (zaheer-Khan3260) [reply client/package-lock.json:1]: sorry for this i remove this file

File: client/yarn.lock
Comment (zaheer-Khan3260) [reply client/yarn.lock:1]: Okay

File: client/src/pages/admin/AddJudges.tsx (hunk 1 above)
Comment (MichaelZhao21) [reply client/src/pages/admin/AddJudges.tsx:19]: This is such an unintuitive way of triggering a re-render in a component. Please use the global store.

Comment (zaheer-Khan3260) [reply client/src/pages/admin/AddJudges.tsx:19]: ok

**Comment by zaheer-Khan3260** [comment]
> This does not solve the problem at hand. You should not be passing callbacks to children. The stats also do not update with these changes. The only change that was needed was a call to the `fetchJudges` function in the Zustand store. There is no need to add more code than one line. You should also not be changing or adding any of the lockfiles.
> 
> Furthermore, you did not use the provided pull request template. This will not be accepted in the future.

Sorry for the misunderstanding but can you please tell from where i need to call  `fetchjudges` function so i can get the updated judges count.
